scripts/dev-environment.sh

```
@@ -571,29 +629,52 @@ show_status() {
-    if [ $MINING_PID -ne 0 ] && kill -0 $MINING_PID 2>/dev/null; then
-        echo "✅ Mining Loop: Running (PID: $MINING_PID)"
+    local mining_pid=""
+    if [ -f ".dev-mining-loop.pid" ]; then
+        local pid_from_file=$(cat .dev-mining-loop.pid 2>/dev/null | head -1 || true)
+        if [ -n "$pid_from_file" ] && kill -0 $pid_from_file 2>/dev/null; then
+            mining_pid=$pid_from_file
+        fi
+    fi
+
+    if is_bitcoin_running; then
+        local block1=$(bitcoin_cli getblockcount 2>/dev/null)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)